fud/fud/stages/xilinx/execution.py

```diff
@@ -75,21 +60,56 @@ def run(xclbin: SourceType.Path) -> SourceType.String:
                         f"user_pre_sim_script={new_dir.name}/pre_sim.tcl\n"
                     )
                     xrt_ini_config.append(
-                        f"user_post_sim_script={new_dir.name}/pre_sim.tcl\n"
+                        f"user_post_sim_script={new_dir.name}/post_sim.tcl\n"
                     )
                 f.writelines(xrt_ini_config)
 
             # Extra Tcl scripts to produce a VCD waveform dump.
             if waveform:
                 with open("pre_sim.tcl", "w") as f:
-                    f.writelines([
-                        "open_vcd\n",
-                        "log_vcd *\n",
-                    ])
+                    f.writelines(
+                        [
+                            "open_vcd\n",
+                            "log_vcd *\n",
+                        ]
+                    )
                 with open("post_sim.tcl", "w") as f:
-                    f.writelines([
-                        "close_vcd\n",
-                    ])
+                    f.writelines(
+                        [
+                            "close_vcd\n",
+                        ]
+                    )
+
+            os.environ["XRT_INI_PATH"] = f"{new_dir.name}/xrt.ini"
+
+        @builder.step()
+        def import_libs():
+            """Import optional libraries"""
+            try:
+                from fud.stages.xilinx import fud_pynq_script
+
+                self.pynq_script = fud_pynq_script
+            except ImportError:
+                raise errors.RemoteLibsNotInstalled
+
+        @builder.step()
+        def run(xclbin: SourceType.Path) -> SourceType.String:
+            """Run the xclbin with datafile"""
+
+            if data_path is None:
+                raise errors.MissingDynamicConfiguration("fpga.data")
+            # Go up a directory due to config going down
+            abs_data_path = Path("../" + data_path).resolve()
+            xclbin = Path("..") / xclbin
+            abs_xclbin_path = xclbin.resolve()
```

Comment:
I get why this is going on here, but using `..` to go up a directory seems somewhat dicey to me: it seems like this will work in the `save_temps` case but perhaps not when `save_temps` is disabled (because the working directory will now be somewhere random in `/tmp`).

Perhaps an alternative would be to just convert `data_path` and `xclbin` to absolute paths at the top level (outside of any particular step, at the very top of this `_define_steps` method)?

Reply:
I think the most recent commit fixes this. In testing works to both create vcd files and emulate without saving any waveforms.
At lines 38-43 and 103-108 in changed file (can't seem to hyperlink to them)